phys2bids/conversion.py

```diff
@@ -0,0 +1,310 @@
+"""
+1. Extract segments from physio file.
+    - Each segment must be named according to the onset time if there are
+      multiple segments. If there is only one segment, then there is no need
+      for a timestamp.
+2. Extract trigger periods from physio segments.
+    - Onset
+3. Extract scan times
+    - Name
+    - Onset with subsecond resolution as close to trigger pulse as possible
+    - Duration (from TR * data dimension)
+4. Calculate difference in time between each scan and each physio trigger.
+5. Use differences to identify similar values across as many physio/scan pairs as possible.
+    - Physio may be missing if trigger failed.
+    - Physio may be delayed if task was manually trigger after scan began
+      (sometimes happens with resting-state scans where the task itself isn't
+      very important).
+    - Scan may be missing if it wasn't converted (e.g., a scan stopped early
+      and re-run).
+6. Assign scan names to trigger period, infer times for other scan times
+   in cases where trigger failed, and ignore trigger periods associated with
+   scans that weren't kept in the BIDS dataset.
+"""
+import os.path as op
+
+from bids import BIDSLayout
+import pandas as pd
+import numpy as np
+import nibabel as nib
+from datetime import datetime
+
+
+def extract_physio_onsets(physio_file):
+    """
+    Collect onsets from physio file, both in terms of seconds and time series
+    indices.
+    TODO: Replace file-loading with phys2bids physio_obj
+    TODO: Stitch segments together before extracting onsets.
+    """
+    import bioread
+    from operator import itemgetter
+    from itertools import groupby
+    physio_data = bioread.read_file(physio_file)
+    trigger_channel = physio_data.channels[-1]
```

Comment:
One of the things on my to-do list is to use the native `phys2bids` classes. I don't know much about them at the moment, so that may take some time, but I believe that selecting the trigger channel is one of the possible parameters for the main physio container object.

That said, if your interest is more in making it easier to test/debug for the near future, I'm happy to make the trigger channel selection more flexible.